xbout/geometries.py

```diff
@@ -74,50 +79,80 @@ def wrapper(add_geometry_coords):
     return wrapper
 
 
+def _set_default_toroidal_coordinates(coordinates):
+    if coordinates is None:
+        coordinates = {}
+
+    # Replace any values that have not been passed in with defaults
+    coordinates['x'] = coordinates.get('x', 'psi')
+    coordinates['y'] = coordinates.get('y', 'theta')
+    coordinates['z'] = coordinates.get('z', 'phi')
+
+
 @register_geometry('toroidal')
-def add_toroidal_geometry_coords(ds):
+def add_toroidal_geometry_coords(ds, coordinates=None):
+
+    coordinates = _set_default_toroidal_coordinates(coordinates)
+
+    # Check whether coordinates names conflict with variables in ds
+    bad_names = []
```

Comment:
You could have saved lines with a list comprehension:
`bad_names = [name for name in coordinates.values() if name in ds]`
but it isn't really any clearer so happy either way.

Reply:
I think the list comprehension is nicer... will update :+1: